lib/management.js

```diff
@@ -0,0 +1,424 @@
+    async.eachSeries(conf.replicationStreams, configureStream, err => {
+        if (err) {
+            return cb(err);
+        }
+        currentState.overlayVersion = conf.version;
+        async.eachSeries(deletedStreamIds, deleteStream, err => {
+            if (err) {
+                return cb(err);
+            }
+            return saveCurrentReplicationState(db, currentState, cb);
+        });
+        return undefined;
+    });
+    return undefined;
+}
+
+function saveServiceCredentials(conf, auth) {
+    // const instanceAuth = JSON.parse(auth);
+    const instanceAuth = auth;
+    const serviceAccount = (conf.users || []).find(
+        u => u.accountType === 'service-replication');
+    if (!serviceAccount) {
+        return;
+    }
+
+    // TODO generate a proper canonicalID during the service account /
+    // instance provisioning
+    // TODO use a proper account id in arn
+
+    const secretKey = decryptSecret(instanceAuth, serviceAccount.secretKey);
+    serviceCredentials = {
+        accounts: [{
+            name: 'service-replication',
+            arn: 'aws::iam:234456789012:root',
+            // canonicalID:
+// '79a59df900b949e55d96a1e698fbacedfd6e09d98eacf8f8d5218e7cd47ef2be',
+            canonicalID: '12349df900b949e55d96a1e698fbacedfd6e09d98eacf8f8d52',
+            displayName: serviceAccount.userName,
+            keys: {
+                access: serviceAccount.accessKey,
+                secret: secretKey,
+            },
+        }],
+    };
+    // initialized = true;
+}
+
+function loadOverlayVersion(metadata, version, cb) {
+    metadata.getObjectMD(serviceBucket, `configuration/overlay/${version}`, {},
+    logger, (err, val) => {
+        if (err) {
+            return cb(err);
+        }
+        return cb(null, val);
+    });
+}
+
+function patchConfiguration(conf, cb) {
+    if (conf.version === undefined) {
+        return process.nextTick(cb, null, conf);
+    }
+    if (overlayVersion !== undefined &&
+        overlayVersion >= conf.version) {
+        return process.nextTick(cb, null, conf);
+    }
+    if (conf && conf.locations) {
+        try {
+            const locations = conf.locations;
+            const locationsWithReplicationBackend = Object.keys(locations)
+            // NOTE: In Orbit, we don't need to have Scality location in our
+            // destination bootstrapList config, since we do not replicate to
+            // any Scality Instance yet.
+            .filter(key => locations[key].locationType !== 'location-file-v1')
+            .reduce((obj, key) => {
+                /* eslint no-param-reassign:0 */
+                obj[key] = locations[key];
+                return obj;
+            }, {});
+            config.setBootstrapList(locationsWithReplicationBackend);
+        } catch (error) {
+            return cb(error);
+        }
+        overlayVersion = conf.version;
+    }
+    return process.nextTick(cb, null, conf);
+}
+
+function initManagement(done) {
```

Comment:
It could be necessary to have a REMOTE_MANAGEMENT_DISABLED env var similar to S3 one in Federation deployments.